### PR TITLE
DerivativeForm output operator: Add test and changelog

### PR DESCRIPTION
**Pull request by bergbauer**
As requested in https://github.com/dealii/dealii/pull/16082#issuecomment-1744794378

### CODE REVIEW

**Review by kronbichler — COMMENTED** [review]
Doesn't the comment you added say that we have additional spaces between the outer dimensions of the tensor? This is not the case in the given output as I could also understand from the code and, in fact, also not `Tensor`. Do you want to update the comments? I could see some value in having extra spaces because I always end up counting by hand, but I think that would break previous behavior.

**Comment by bergbauer** [comment]
> Doesn't the comment you added say that we have additional spaces between the outer dimensions of the tensor? This is not the case in the given output as I could also understand from the code and, in fact, also not Tensor. Do you want to update the comments? I could see some value in having extra spaces because I always end up counting by hand, but I think that would break previous behavior.

I can also add the extra spaces. as far as I can see this is not tested. (I also see the value in spaces. I just tested it.) numdiff also does not care about the spaces, tests are passing anyways.

Modified output:
```
DEAL::dF : 1.00000
DEAL::ddF: 1.00000
DEAL::dF : 1.00000  2.00000
DEAL::ddF: 1.00000   2.00000
DEAL::dF : 1.00000  2.00000  3.00000
DEAL::ddF: 1.00000   2.00000   3.00000
DEAL::dF : 1.00000 3.00000  2.00000 4.00000
DEAL::ddF: 1.00000 4.00000  3.00000 6.00000   2.00000 5.00000  4.00000 7.00000
DEAL::dF : 1.00000 3.00000  2.00000 4.00000  3.00000 5.00000
DEAL::ddF: 1.00000 4.00000  3.00000 6.00000   2.00000 5.00000  4.00000 7.00000   3.00000 6.00000  5.00000 8.00000
DEAL::dF : 1.00000 3.00000 5.00000  2.00000 4.00000 6.00000  3.00000 5.00000 7.00000
DEAL::ddF: 1.00000 4.00000 7.00000  3.00000 6.00000 9.00000  5.00000 8.00000 11.0000   2.00000 5.00000 8.00000  4.00000 7.00000 10.0000  6.00000 9.00000 12.0000   3.00000 6.00000 9.00000  5.00000 8.00000 11.0000  7.00000 10.0000 13.0000
```

File: tests/base/derivative_forms_05.cc [hunk 1]
```diff
@@ -0,0 +1,58 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2000 - 2018 by the deal.II authors
```
Comment (bangerth) [inline tests/base/derivative_forms_05.cc:3]:
```suggestion
// Copyright (C) 2023 by the deal.II authors
```

**Review by bangerth — APPROVED** [review]
Thanks for this follow-up. I agree that it would be nice to add the spaces, since that's what the documentation already says. Given that numdiff doesn't seem to care, let's merge this as-is.

**Comment by bangerth** [comment]
Actually, given the copyright years thing, and since the tests aren't finished yet, I'll let you decide if you want to do the spaces here or in a separate patch.

**Comment by bergbauer** [comment]
Let's fix it here. This is how I would expect the spaces to look like. Do you agree? @bangerth @kronbichler

File: tests/base/derivative_forms_05.cc [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+// test operator<<
```
Comment (peterrum) [inline tests/base/derivative_forms_05.cc:16]: of what?

**Review by peterrum — APPROVED** [review]
Looks reasonable.

**Review by kronbichler — APPROVED** [review]
I like this, thank you! It turned out to be less complicated than I feared. :+1:

File: tests/base/derivative_forms_05.cc (hunk 2 above)
Comment (bergbauer) [reply tests/base/derivative_forms_05.cc:16]: I thought that was obvious, given the file name. I can update the comment.

Comment (bergbauer) [reply tests/base/derivative_forms_05.cc]: Updated!

**Review by peterrum — APPROVED** [review]

**Pull request merged**